src/types/slice.rs

```diff
@@ -61,6 +61,7 @@ impl PySlice {
             let start: isize = 0;
             let stop: isize = 0;
             let step: isize = 0;
+            #[allow(deprecated)]
             let r = ffi::PySlice_GetIndicesEx(
```

Comment:
What's the recommended replacement to using `PySlice_GetIndicesEx`? It would be great to remove this usage of a deprecated API...

Reply:
In cpython, they define a macro `PySlice_GetIndicesEx` (shown below)

```c
#define PySlice_GetIndicesEx(slice, length, start, stop, step, slicelen) (  \
    PySlice_Unpack((slice), (start), (stop), (step)) < 0 ?                  \
    ((*(slicelen) = 0), -1) :                                               \
    ((*(slicelen) = PySlice_AdjustIndices((length), (start), (stop), *(step))), \
     0))
```

Which I didn't implement since currently we didn't have `PySlice_Unpack` and `PySlice_AdjustIndices`.
These method seems not so hard. If you like, I can also implement these.

Reply:
That would be great, yes please!

Given the function is deprecated in Python 3.7, and there's [quite a complex condition on the macro, `PySlice_Unpack` and `PySlice_AdjustIndices` which suggest they were added inPython 3.6.1](https://github.com/python/cpython/blob/5277ffe12d492939544ff9c54a3aaf448b913fb3/Include/sliceobject.h#L49), I think we can only use the new macro form in Python 3.7 and up.